moveos/moveos-stdlib/src/natives/moveos_stdlib/move_module.rs

```diff
@@ -265,6 +268,123 @@ fn check_compatibililty_inner(
+    debug_assert!(args.len() == 3, "Wrong number of arguments");
+    let mut cost = gas_params.base;
+    let new_address_vec = pop_arg!(args, Vector);
+    let old_address_vec = pop_arg!(args, Vector);
+    let num_addresses = new_address_vec.elem_views().len();
+    if num_addresses != old_address_vec.elem_views().len() {
+        return Ok(NativeResult::err(
+            cost,
+            moveos_types::move_std::error::invalid_argument(E_LENTH_NOT_MATCH),
+        ));
+    };
+    let num_addresses = num_addresses as u64;
+    let new_addresses = new_address_vec.unpack(&Type::Address, num_addresses)?;
+    let old_addresses = old_address_vec.unpack(&Type::Address, num_addresses)?;
+
+    let address_mapping: HashMap<AccountAddress, AccountAddress> =
+        zip_eq(old_addresses, new_addresses)
+            .map(|(a, b)| {
+                Ok((
+                    a.value_as::<AccountAddress>()?,
+                    b.value_as::<AccountAddress>()?,
+                ))
+            })
+            .collect::<PartialVMResult<_>>()?;
+
+    let mut bundle = vec![];
+    for module in pop_arg!(args, Vec<Value>) {
+        let byte_codes = module.value_as::<Vec<u8>>()?;
+        cost += gas_params.per_byte * NumBytes::new(byte_codes.len() as u64);
+        bundle.push(byte_codes);
+    }
+    let mut compiled_modules = bundle
+        .into_iter()
+        .map(|b| CompiledModule::deserialize(&b))
+        .collect::<PartialVMResult<Vec<CompiledModule>>>()?;
+
+    let mut remapped_bubdles = vec![];
+    for m in compiled_modules.iter_mut() {
+        // TODO: charge gas
+        module_remap_addresses(m, &address_mapping)?;
+        let mut binary: Vec<u8> = vec![];
+        m.serialize(&mut binary).map_err(|e| {
+            PartialVMError::new(StatusCode::VALUE_SERIALIZATION_ERROR).with_message(e.to_string())
+        })?;
+        let value = Value::vector_u8(binary);
+        remapped_bubdles.push(value);
+    }
+    let output_modules = Vector::pack(&Type::Vector(Box::new(Type::U8)), remapped_bubdles)?;
+    Ok(NativeResult::ok(cost, smallvec![output_modules]))
+}
+
+fn module_remap_constant_addresses(value: &mut MoveValue, f: &dyn Fn(&mut AccountAddress)) {
+    match value {
+        MoveValue::Address(addr) => f(addr),
+        MoveValue::Vector(vals) => {
+            vals.iter_mut()
+                .for_each(|val| module_remap_constant_addresses(val, f));
+        }
+        // TODO: handle constant addresses in Other struct
+        _ => {}
+    }
+}
+
+fn module_remap_addresses(
+    module: &mut CompiledModule,
+    address_mapping: &HashMap<AccountAddress, AccountAddress>,
+) -> PartialVMResult<()> {
+    // replace addresses in address identifiers.
+    for addr in module.address_identifiers.iter_mut() {
+        if let Some(new_addr) = address_mapping.get(addr) {
+            *addr = *new_addr;
+        }
+    }
+    // replace addresses in constant.
+    for constant in module.constant_pool.iter_mut() {
```

Comment:
Should the address in constant_pool be replaced? I think remap only changes the module address and should not change the constant.

Reply:
Or we provide low-level API to the developer, such as `replace_constant` and `replace_address_identifier`, `replace_identifier`, let the developer decide based on the use-case. And `replace_identifier` also can resolve the #1002